### PR TITLE
refactor: replace `globby` with `tinyglobby`

### PR DESCRIPTION
**Pull request by SuperchupuDev**
supersedes #1167, does basically the same but with globbing logic moved into [its own library](https://github.com/SuperchupuDev/tinyglobby) as suggested by maintainers

> this is the followup pr mentioned in #1163. i've opened this mainly to get feedback from the maintainers on the best way to implement this. i can port the globbing code into its own package if needed, even though this isn't that much code
>
> according to [globby's readme](https://github.com/sindresorhus/globby/blob/main/readme.md#features) the only exclusive features it has (that are relevant in this case) are negated patterns and directory expansion, which this pr both covers
>
> the issues reported in 6.2.1 were all related to either me forgetting to handle negated patterns in `entry` or not knowing directory expansion was meant to be done. if there is any more globby-exclusive functionality missing let me know and i'll add it
> 
> i'd be happy to test any possible edge cases in the logic and add new tests

ref: https://github.com/egoist/tsup/pull/1163#issuecomment-2242438664

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -57,9 +57,9 @@ export async function run(
   }
 
   // Get output
-  const outFiles = await glob('**/*', {
+  const outFiles = await glob({
     cwd: path.resolve(testDir, 'dist'),
-  }).then((res) => res.sort())
+  }).then((res) => res.map((f) => f.replaceAll('\\', '/')).sort())
```
Comment (sxzz) [inline test/utils.ts:62]: Use `slash` util.

**Review by sxzz — APPROVED** [review]
